presto-mysql/src/main/java/com/facebook/presto/plugin/mysql/MySqlClient.java

```diff
@@ -111,10 +115,24 @@ protected SchemaTableName getSchemaTableName(ResultSet resultSet)
     @Override
     protected String toSqlType(Type type)
     {
+        if (type instanceof VarcharType) {
+            VarcharType varcharType = (VarcharType) type;
+            if (varcharType.getLength() <= MAX_IN_ROW_VARCHAR_LENGTH) {
+                return "varchar(" + varcharType.getLength() + ")";
+            }
+            else if (varcharType.getLength() <= 65535) {
+                return "text";
```

Comment:
Hmm.. this may be problematic. "text" is supposed to be <= 64k **bytes**, but the length parameter for varchar indicates number of unicode characters.